local.c

```diff
@@ -835,6 +835,24 @@ static int add_attr_to_channel(struct iio_channel *chn,
 	union iio_pointer p = { .chn = chn, };
 	struct iio_attr_list *attrs;
 	int ret;
+	size_t lbl_len = strlen("_label");
+	const char *chn_label_ptr = NULL;
+	char label[512];
+	const char *dev_id;
+
+	if (strlen(name) >= lbl_len &&
+		!strcmp(name + strlen(name) - lbl_len, "_label")) {
+		dev_id = iio_device_get_id(iio_channel_get_device(chn));
+		ret = local_do_read_dev_attr(dev_id, 0, name,
+			label, sizeof(label), IIO_ATTR_TYPE_CHANNEL);
+		if (ret > 0) {
+			chn_label_ptr = label;
+			chn->label = iio_strdup(chn_label_ptr);
```

Comment:
can this simply be `chn->label = iio_strdup(label);`?

Reply:
it can be. Thanks!